kernel/sys/tasks.cpp

```diff
@@ -143,7 +197,11 @@ static void _task_starting()
     // it is run in the context of the new task
 
     // the task before this caused the scheduler to lock
-    _release_scheduler_lock();
+    // so we must unlock here
+    _scheduler_lock--;
```

Comment:
Why are you calling `--` on a lock? We don't have an operator overwritten for that, do we?

Reply:
Nope. It's just a plain integer. It's decremented because it's a recursive lock.

Reply:
Sounds good.